x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/action/util/ExpandedIdsMatcher.java

```diff
@@ -137,6 +137,49 @@ public boolean isOnlyExact() {
         return onlyExact;
     }
 
+
+    /**
+     * A simple matcher with one purpose to test whether an id
+     * matches a expression that may contain wildcards.
+     * Use the {@link #idMatches(String)} function to
+     * test if the given id is matched by any of the matchers.
+     *
+     * Unlike {@link ExpandedIdsMatcher} there is no
+     * allowNoMatchForWildcards logic and the matchers
+     * are not be removed once they have been matched.
+     */
+    public static class SimpleIdsMatcher {
```

Comment:
Why do we need this? Can't we use `ExpandedIdsMatcher` instead?

Reply:
In the first pass I used `ExpandedIdsMatcher` and added the `idMatches(Id)` but this meant I had to add a lot of javadoc explaining that the `idMatches()` method is not compatible with calling `filterMatchedIds()` as `filterMatchedIds()` removes matchers so you will get different results for the same input if `filterMatchedIds()` is called inbetween.

In the case where the token is `*` `ExpandedIdsMatcher` will only match an input that if `allowNoMatchForWildcards == false` as otherwise it does not create the wild card matcher.  If   `allowNoMatchForWildcards == true` then `idMatches()` will never match as there is no wild card matcher. 

So what with all the Javadoc and the fact that it required knowledge of the class internals to get the behaviour I wanted I thought it best to create a new explicit class that does one thing and let `ExpandedIdsMatcher` do it's own thing.

That said I can do better with naming. How about I renamed this file `IdMatchers` and have 2 nested classes `IdMatchers.RequiredIdsMatcher()` and  `IdMatchers.SimpleIdsMatcher()`